zerver/lib/import_realm.py

```diff
@@ -504,10 +496,17 @@ def re_map_foreign_keys_internal(
             item[field_name + "_id"] = new_id
             del item[field_name]
         else:
-            if reaction_field:
-                item[field_name] = str(new_id)
-            else:
-                item[field_name] = new_id
+            item[field_name] = new_id
+
+
+def re_map_realm_emoji_codes(data: TableData, *, table_name: str) -> None:
+    emoji_code_dct = {}
+    for row in data["zerver_realmemoji"]:
+        emoji_code_dct[row["name"]] = str(row["id"])
+
+    for row in data[table_name]:
+        if row["reaction_type"] == Reaction.REALM_EMOJI:
+            row["emoji_code"] = emoji_code_dct[row["emoji_name"]]
```

Comment:
I don't think this algorithm is correct in the presence of deactivated custom emoji, which make it possible to have multiple custom emoji objects with the same name.  (The rule is only 1 can be non-deactivated).

I think we just need to return to the ID-based algorithms that doesn't have this bug.  (Also we should just generally avoid using name type fields when an ID is present).

Reply:
FWIW I also don't think the `reaction_field` here is a hack: we just have a column where `emoji_code` is a foreign key to RealmEmoji only for the `realm_emoji` reaction type, and we were just representing that conditional foreign key arrangement.